server/db/models/order.js

```diff
@@ -10,19 +10,46 @@ const Order = db.define('order', {
     type: Sequelize.ENUM('Cart', 'Cancelled', 'Completed'),
     allowNull: false,
     defaultValue: 'Cart'
+  },
+  isShipped: {
+    type: Sequelize.BOOLEAN,
+    defaultValue: false,
+    allowNull: false
   }
 })
 
-module.exports = Order
-
 /**
  * instanceMethods
  */
-
+Order.prototype.checkout = function() {
+  if (this.isCart) {
+    this.isCart = false
+    this.status = 'Processing'
+    let updatedProducts = this.products.map(product => {
```

Comment:
I have question on Line 28, since we don't have products as property define on order model, would "this.products" returned undefined?